### PR TITLE
Workshop revision: Debugging JS3 Week 3

### PR DESCRIPTION
**Pull request by SallyMcGrath**
This workshop takes the original debugging folder and extends to be a bit more directed.

The conceit here is to also play computer and see how the debugger relates to this and also the pythontutor visualiser. 

I've provided all the original problems as a bank. I've also added a vscode launch settings which I think makes it work on any file but hm, I'm not super sure to be honest. Would really appreciate some eyes on this.

The core workshop can be done in 45 minutes and it has extensions if people are into it.

Please review and merge as this PR is blocking https://github.com/CodeYourFuture/curriculum/pull/400

### CODE REVIEW

**Review by 40thieves — APPROVED** [review]
Looks good!

I left a quick suggestion and have some questions about teaching the various slightly different interfaces of debuggers, but happy to merge this to get it moving along.

**Comment by 40thieves** [comment]
I've forgotten why we enforce 2 reviewers? Don't think this will cause issues so will bypass and hope for forgiveness later.

**Comment by SallyMcGrath** [comment]
Thanks @40thieves  ❤️ 

We have two reviewers on this repo purely because trainees often open PRs to it. Students team only have Read access but there are some activities where they are asked to merge to a fork. The 2 reviews is purely to offer a brake so that helpful Mentors team (who do have write access) don't merge in here for them.  

It's a bit annoying but, yeah, all Syllabus team members also have Powers and can make the correct decision to bypass in a time crunch.

File: debugging/readme.md
Comment (SallyMcGrath) [reply debugging/readme.md:6]: Ahhh, so the link is wrong, I've just realised. I definitely thought I was linking https://code.visualstudio.com/docs/introvideos/debugging

**Pull request merged**
